core/src/main/scala/org/broadinstitute/dsde/rawls/fastpass/FastPassService.scala

```diff
@@ -0,0 +1,444 @@
+                                   googleProject: GoogleProject
+  ): Future[Unit] = {
+    logger.info(
+      s"Removing FastPass IAM bindings for ${workbenchEmail.value} on ${resourceType.value} $resourceName"
+    )
+    for {
+      _ <- resourceType match {
+        case IamResourceTypes.Bucket =>
+          googleStorageDAO.removeIamRoles(
+            GcsBucketName(resourceName),
+            workbenchEmail,
+            memberType,
+            organizationRoles,
+            userProject = Some(googleProject)
+          )
+        case IamResourceTypes.Project =>
+          googleIamDao.removeRoles(
+            googleProject,
+            workbenchEmail,
+            memberType,
+            organizationRoles
+          )
+      }
+      _ <- openTelemetry
+        .incrementCounter(s"fastpass-iam-revoked-${memberType.value}", tags = openTelemetryTags)
+        .unsafeToFuture()
+    } yield ()
+  }
+
+  private def setupProjectRoles(workspace: Workspace,
+                                samResourceRoles: Set[SamResourceRole],
+                                userAndPet: UserAndPetEmails,
+                                expirationDate: DateTime
+  ): ReadWriteAction[Unit] = {
+    val projectIamRoles = samResourceRoles.flatMap(samWorkspaceRoleToGoogleProjectIamRoles)
+    val condition = conditionFromExpirationDate(expirationDate)
+
+    for {
+      _ <- DBIO.from(addUserAndPetToProjectIamRoles(workspace.googleProjectId, projectIamRoles, userAndPet, condition))
+      _ <- writeGrantsToDb(
+        workspace.workspaceId,
+        userAndPet,
+        gcpResourceType = IamResourceTypes.Project,
+        workspace.googleProjectId.value,
+        projectIamRoles,
+        expirationDate
+      )
+    } yield ()
+  }
+
+  private def setupBucketRoles(workspace: Workspace,
+                               samResourceRoles: Set[SamResourceRole],
+                               userAndPet: UserAndPetEmails,
+                               expirationDate: DateTime
+  ): ReadWriteAction[Unit] = {
+    val bucketIamRoles = samResourceRoles.flatMap(samWorkspaceRolesToGoogleBucketIamRoles)
+    val condition = conditionFromExpirationDate(expirationDate)
+
+    for {
+      _ <- DBIO.from(
+        addUserAndPetToBucketIamRole(GcsBucketName(workspace.bucketName),
+                                     bucketIamRoles,
+                                     userAndPet,
+                                     condition,
+                                     workspace.googleProjectId
+        )
+      )
+      _ <- writeGrantsToDb(
+        workspace.workspaceId,
+        userAndPet,
+        gcpResourceType = IamResourceTypes.Bucket,
+        workspace.bucketName,
+        bucketIamRoles,
+        expirationDate
+      )
+    } yield ()
+  }
+
+  private def writeGrantsToDb(workspaceId: String,
+                              userAndPet: UserAndPetEmails,
+                              gcpResourceType: IamResourceType,
```

Comment:
Naming inconsistency.  I personally think `GcpResourceType` is better than `IamResourceType`, since that's really what we're referring too and that enum might be more useful if named as such.